PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/coreTools/FunctionsCoreToolsManager.kt

```diff
@@ -211,32 +211,40 @@ object FunctionsCoreToolsManager {
     }
 
     fun determineLatestRemote(): AzureFunctionsCoreToolsRemoteAsset? {
+        val expectedFileNamePrefix = "Azure.Functions.Cli." + if (SystemInfo.isWindows && SystemInfo.is64Bit) {
+            "win-x64"
+        } else if (SystemInfo.isWindows) {
+            "win-x86"
+        } else if (SystemInfo.isMac) {
+            "osx-x64"
+        } else if (SystemInfo.isLinux) {
+            "linux-x64"
+        } else {
+            "unknown"
+        }
+
         try {
-            val latestGitHubRelease = GitHubReleasesService.createInstance()
-                    .getLatestRelease(LATEST_RELEASE_URL)
+            val gitHubReleases = GitHubReleasesService.createInstance()
+                    .getReleases(API_URL_RELEASES)
                     .execute()
                     .body()
 
-            if (latestGitHubRelease != null) {
-                val expectedFileNamePrefix = "Azure.Functions.Cli." + if (SystemInfo.isWindows && SystemInfo.is64Bit) {
-                    "win-x64"
-                } else if (SystemInfo.isWindows) {
-                    "win-x86"
-                } else if (SystemInfo.isMac) {
-                    "osx-x64"
-                } else if (SystemInfo.isLinux) {
-                    "linux-x64"
-                } else {
-                    "unknown"
-                }
+            if (gitHubReleases != null) {
+                for (gitHubRelease in gitHubReleases
+                        .filter { !it.tagName.isNullOrEmpty() }
+                        .sortedWith(Comparator { o1, o2 ->
+                            VersionComparatorUtil.compare(o2.tagName!!.trimStart('v'), o1.tagName!!.trimStart('v')) // latest versions on top
```

Comment:
I would not 100% rely on Release naming based on tags :)

Reply:
Since there's no other way of determining actual version number I think this is the next best thing :-/